platforminfo/platform_info_linux.go

```diff
@@ -471,27 +471,40 @@ func GetCBNTHardwareFeature() (CBNT, error) {
 		return cbnt, err
 	}
 
-	// check if CBNT is enabled
+	// Check if CBNT is supported
 	// similar to 'rdmsr -f 32:32 0x13A
-	cbntEnabled, err := GetBits(cbntBits, 32, 32)
+	cbntSupported, err := GetBits(cbntBits, 32, 32)
 	if err != nil {
 		return cbnt, err
 	}
 
-	// CBNT is enabled, create a CBNT structure and populate it.
-	if cbntEnabled == 1 {
-		cbnt.Meta.Profile, err = GetCBNTProfile(cbntBits)
+	// CBNT is supported, create a CBNT structure and populate it.
+	if cbntSupported == 1 {
+		cbnt.HardwareFeature.Supported = true
+		bitsVMF, err := GetBits(cbntBits, 7, 4)
 		if err != nil {
 			return cbnt, err
 		}
-		cbnt.Enabled = cbnt.Meta.Profile != CBNT_PROFILE_0
-		cbnt.Meta.MSR = CBNT_PROCESSOR_FLAGS
+
+		bitBTGEnabled, err := GetBits(cbntBits, 0, 0)
+		if err != nil {
+			return cbnt, err
+		}
+
+		if bitsVMF != 0 || bitBTGEnabled != 0 {
+			cbnt.Meta.Profile, err = GetCBNTProfile(cbntBits)
+			if err != nil {
+				return cbnt, err
+			}
+			cbnt.Enabled = cbnt.Meta.Profile != CBNT_PROFILE_0
+			cbnt.Meta.MSR = CBNT_PROCESSOR_FLAGS
+		}
 	}
```

Comment:
shouldn't there be else block with "cbnt.HardwareFeature.Supported = false"?

Reply:
It is not needed, since the default value for the CBNT structure members will be setting as false only. Verified with the host machine which is not having CBNT and the supported value is false only.